lib/MongoDB/Role/_SessionSupport.pm

```diff
@@ -0,0 +1,95 @@
+#
+#  Copyright 2014 MongoDB, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+use strict;
+use warnings;
+package MongoDB::Role::_SessionSupport;
+
+# MongoDB role to add support for sessions on Ops
+
+use Moo::Role;
+use MongoDB::_Types -types, 'to_IxHash';
+use Scalar::Util qw/ blessed /;
+use Types::Standard qw(
+    Maybe
+    InstanceOf
+);
+use namespace::clean;
+
+requires qw/ session /;
+
+# If theres no session, but there is a cluster time, how do we get it from the
+# client?
```

Comment:
I think if sessions are supported, then we should have either an implicit or explicit session or we're in some weird edge case that doesn't allow sessions and we can probably safely skip messing with clustertime in that case as well.